python/lsst/daf/butler/formatters/parquet.py

```diff
@@ -136,6 +136,8 @@ def write(self, inMemoryDataset: Any) -> None:
             arrow_table = astropy_to_arrow(inMemoryDataset)
         elif isinstance(inMemoryDataset, np.ndarray):
             arrow_table = numpy_to_arrow(inMemoryDataset)
+        elif isinstance(inMemoryDataset, dict):
+            arrow_table = numpy_dict_to_arrow(inMemoryDataset)
```

Comment:
Interesting question.  If it isn't it will crash, but maybe with a less helpful error than the unsupported type `ValueError`.  I'd like to know @timj's thoughts on this.

Reply:
One downside of using a generic python type is that butler can't help you very much since it's not looking inside it. You could put a try block around this if you wanted to reraise a ValueError...